### DISCUSSION TITLE
One to many relationship

### DISCUSSION THREAD

**Discussion by ltsrc** [prisma/prisma#5747] [Q&A] [answered]
Hello, I am currently trying to figure solutions to this problem: I have multiple models that have a one-to-many relationship with a model called `Purchases`. So the schema will be:

```prisma
model A {
    id        Int        @id @default(autoincrement())
    purchases Purchase[]
}

model B {
    id        Int        @id @default(autoincrement())
    purchases Purchase[]
}

model C {
    id        Int        @id @default(autoincrement())
    purchases Purchase[]
}

model Purchase {
    id  Int  @id @default(autoincrement())

    // Default behavior
    A   A?   @relation(fields: [aId], references: [id])
    aId Int?
    B   B?   @relation(fields: [bId], references: [id])
    bId Int?
    C   C?   @relation(fields: [cId], references: [id])
    cId Int?
}
```
But the thing is I might be creating a lot of new models (A, B, C  etc) and I find it problematic to keep an id for each of the relation. One trick I've found is to use a many-to-many relationship, which reduces the schema to:

```prisma
model Purchase {
    id  Int  @id @default(autoincrement())

    A   A[]
    B   B[]
    C   C[]
}
```

But I don't like this as it is a trick and as a Purchase must only be linked to one model (A | B | C). So I wondered if there was any other solution to this situation, and what are your thoughts about the two I exposed ?

**Comment by ryands17** [comment, marked as answer]
@ltsrc 👋 

If you want a 1-many relationship, then this will be the only way to do this. I would suggest adding a 👍 to this request: https://github.com/prisma/prisma/issues/1985 which has a proposal for simplifying the Relation API so that we can set the priority for this :)

**Reply by ltsrc** [reply]
Alright, thanks for the quick answer :+1:

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
